Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -64,45 +63,212 @@ extension PluginTarget {
+                    
+                case .defineBuildCommand(let config, let inputFiles, let outputFiles):
+                    delegate.pluginDefinedBuildCommand(
+                        displayName: config.displayName,
+                        executable: try AbsolutePath(validating: config.executable),
+                        arguments: config.arguments,
+                        environment: config.environment,
+                        workingDirectory: try config.workingDirectory.map{ try AbsolutePath(validating: $0) },
+                        inputFiles: try inputFiles.map{ try AbsolutePath(validating: $0) },
+                        outputFiles: try outputFiles.map{ try AbsolutePath(validating: $0) })
+                    
+                case .definePrebuildCommand(let config, let outputFilesDir):
+                    delegate.pluginDefinedPrebuildCommand(
+                        displayName: config.displayName,
+                        executable: try AbsolutePath(validating: config.executable),
+                        arguments: config.arguments,
+                        environment: config.environment,
+                        workingDirectory: try config.workingDirectory.map{ try AbsolutePath(validating: $0) },
+                        outputFilesDirectory: try AbsolutePath(validating: outputFilesDir))
+
+                case .buildOperationRequest(let subset, let parameters):
+                    delegate.pluginRequestedBuildOperation(subset: .init(subset), parameters: .init(parameters)) {
+                        do {
+                            switch $0 {
+                            case .success(let result):
+                                responder(try HostToPluginMessage.buildOperationResponse(result: .init(result)).toData())
+                            case .failure(let error):
+                                responder(try HostToPluginMessage.errorResponse(error: String(describing: error)).toData())
+                            }
+                        }
+                        catch {
+                            print("couldn't send reply to plugin \(error)")
+                        }
+                    }
+
+                case .testOperationRequest(let subset, let parameters):
+                    delegate.pluginRequestedTestOperation(subset: .init(subset), parameters: .init(parameters)) {
+                        do {
+                            switch $0 {
+                            case .success(let result):
+                                responder(try HostToPluginMessage.testOperationResponse(result: .init(result)).toData())
+                            case .failure(let error):
+                                responder(try HostToPluginMessage.errorResponse(error: String(describing: error)).toData())
+                            }
+                        }
+                        catch {
+                            print("couldn't send reply to plugin \(error)")
```

Comment:
should this use an observability scope to email instead of print?

Reply:
Oh right.  Yes, I meant to fix this before marking ready.  Thanks for calling it out!

Reply:
Actually, I think what should happen here is that the default script handler should probably hold onto the error and go into an error state.  I'll have to look what should happen (the previous code dropped the error, so potentially we could also improve this separately).